src/Security/Security.php

```diff
@@ -209,19 +318,20 @@ public function deleteProfile($id, array $options = [])
      *
      * @param string $scrollId
      * @param array $options (optional) arguments
+     *
      * @return ProfilesSearchResult
-     * @throws \Exception
+     *
+     * @throws InvalidArgumentException
```

Comment:
(as a consequence, `SearchResult` class and its subclases should be refactored to follow the model of js SDK, and it will impact the implementation of `search*` methods....